docs/quickstarts/react-router.mdx

```diff
@@ -0,0 +1,216 @@
+  ]
+
+  export function Layout({ children }: { children: React.ReactNode }) {
+    return (
+      <html lang="en">
+        <head>
+          <meta charSet="utf-8" />
+          <meta name="viewport" content="width=device-width, initial-scale=1" />
+          <Meta />
+          <Links />
+        </head>
+        <body>
+          {children}
+          <ScrollRestoration />
+          <Scripts />
+        </body>
+      </html>
+    )
+  }
+
+  export default function App() {
+    return <Outlet />
+  }
+
+  export function ErrorBoundary({ error }: Route.ErrorBoundaryProps) {
+    let message = 'Oops!'
+    let details = 'An unexpected error occurred.'
+    let stack: string | undefined
+
+    if (isRouteErrorResponse(error)) {
+      message = error.status === 404 ? '404' : 'Error'
+      details =
+        error.status === 404 ? 'The requested page could not be found.' : error.statusText || details
+    } else if (import.meta.env.DEV && error && error instanceof Error) {
+      details = error.message
+      stack = error.stack
+    }
+
+    return (
+      <main className="pt-16 p-4 container mx-auto">
+        <h1>{message}</h1>
+        <p>{details}</p>
+        {stack && (
+          <pre className="w-full p-4 overflow-x-auto">
+            <code>{stack}</code>
+          </pre>
+        )}
+      </main>
+    )
+  }
+  ```
+
+  ### Add `<ClerkProvider>` and Clerk components to your app
+
+  <Include src="_partials/clerk-provider" />
+
+  You must pass `loaderData` to the `<ClerkProvider>` component. This data is provided by the `rootAuthLoader()` function.
+
+  You can control which content signed-in and signed-out users can see with Clerk's [prebuilt control components](/docs/components/overview#what-are-control-components).
+
+  The following example adds `<ClerkProvider>` and creates a header using the following Clerk components:
+
+  - [`<SignedIn>`](/docs/components/control/signed-in): Children of this component can only be seen while **signed in**.
+  - [`<SignedOut>`](/docs/components/control/signed-out): Children of this component can only be seen while **signed out**.
+  - [`<UserButton />`](/docs/components/user/user-button): Shows the signed-in user's avatar. Selecting it opens a dropdown menu with account management options.
+  - [`<SignInButton />`](/docs/components/unstyled/sign-in-button): An unstyled component that links to the sign-in page. In this example, since no props or [environment variables](/docs/deployments/clerk-environment-variables) are set for the sign-in URL, this component links to the [Account Portal sign-in page](/docs/customization/account-portal/overview#sign-in).
+
+  ```tsx {{ filename: 'app/root.tsx' }}
+  // Other imports
+
+  import { ClerkProvider, SignedIn, SignedOut, UserButton, SignInButton } from '@clerk/react-router'
+
+  export default function App({ loaderData }: Route.ComponentProps) {
+    return (
+      <ClerkProvider loaderData={loaderData}>
```

Comment:
still waiting to hear back about passing sign up/in props here

https://clerkinc.slack.com/archives/C01QFRQNHSN/p1733939261701099?thread_ts=1733934231.713979&cid=C01QFRQNHSN

Reply:
[e7aa1d8](https://github.com/clerk/clerk-docs/pull/1760/commits/e7aa1d83d39621110453d39189ab56be319d82b0)